src/Microsoft.OData.Client/Serialization/ODataPropertyConverter.cs

```diff
@@ -66,9 +66,58 @@ internal IEnumerable<ODataProperty> PopulateProperties(object resource, string s
                 }
             }
 
+            // Process non-structured dynamic properties stored in the open type dictionary
+            if (ClientTypeUtil.IsInstanceOfOpenType(resource, this.requestInfo.Model))
+            {
+                PopulateDynamicProperties(resource, serverTypeName, odataProperties);
+            }
+
             return odataProperties;
         }
 
+        /// <summary>
+        /// Populates list of odata properties with non-structured dynamic properties on an open type
+        /// </summary>
+        /// <param name="resource">Instance of the resource which is getting serialized.</param>
+        /// <param name="serverTypeName">The server type name of the entity whose properties are being populated.</param>
+        /// <param name="odataProperties">List of ODataProperty</param>
+        private void PopulateDynamicProperties(object resource, string serverTypeName, List<ODataProperty> odataProperties)
+        {
+            Debug.Assert(resource != null, "resource !=null");

```

Comment:
What happens when in production and these values are null?

Reply:
@paulodero Responded to this [here](https://github.com/OData/odata.net/pull/1712#discussion_r412895515)

Reply:
Got it.